mesonbuild/options.py

```diff
@@ -941,53 +941,26 @@ def sanitize_dir_option_value(self, prefix: str, option: OptionKey, value: T.Any
         # .as_posix() keeps the posix-like file separators Meson uses.
         return value.as_posix()
 
-    def set_value(self, key: T.Union[OptionKey, str], new_value: 'T.Any', first_invocation: bool = False) -> bool:
-        key = self.ensure_and_validate_key(key)
-        if key.name == 'prefix':
-            new_value = self.sanitize_prefix(new_value)
-        elif self.is_builtin_option(key):
-            prefix = self.get_value_for('prefix')
-            assert isinstance(prefix, str)
-            new_value = self.sanitize_dir_option_value(prefix, key, new_value)
-        if key not in self.options:
-            raise MesonException(f'Unknown options: "{key.name}" not found.')
-
-        valobj = self.options[key]
-        old_value = valobj.value
-        changed = valobj.set_value(new_value)
-
-        if valobj.readonly and changed and not first_invocation:
-            raise MesonException(f'Tried to modify read only option {str(key)!r}')
-
-        if key.name == 'prefix' and first_invocation and changed:
-            assert isinstance(old_value, str)
-            self.reset_prefixed_options(old_value, new_value)
-
-        if changed:
-            self.set_dependents(key, new_value)
-
-        return changed
-
-    def set_dependents(self, key: OptionKey, value: str) -> None:
-        if key.name != 'buildtype':
-            return
+    def _set_dependents(self, key: OptionKey, value: str) -> None:
```

Comment:
This method could really benefit from a better name 🙂

Reply:
Maybe it could be even inlined in the caller: AFAICT there is just one

Reply:
done.